pkg/util/labels.go

```diff
@@ -60,39 +60,137 @@ func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels lab
 	return nil
 }
 
+// addUnstructuredNestedLabels adds new label(s) to a nested labels of a single unstructured object
+func addUnstructuredNestedLabels(obj *runtime.Unstructured, labels labels.Set) error {
```

Comment:
Don't add things to util.  Create a sub package for dealing  with this (or use one of the 3 or four existing ones, like pkg/template) and move it there.  Who else is going to use this besides templates?